### ISSUE TITLE
Deeplab Training on TPU failed with "Tensor had NaN values" Error.

### ISSUE THREAD

**Issue by csmovva** [tensorflow/tpu#633]
Training command:
python ~/tpu/models/experimental/deeplab/main.py \
--mode='train' \
--num_shards=8 \
--alsologtostderr=true \
--model_dir=gs://deepadd/logs \
--dataset_dir=gs://deepadd/tfrecord \
--init_checkpoint=gs://deepadd/deeplabv3_pascal_train_aug/model.ckpt \
--model_variant=xception_65 \
--image_pyramid=1. \
--aspp_with_separable_conv=false \
--multi_grid=1 \
--multi_grid=2 \
--multi_grid=4 \
--decoder_use_separable_conv=false \
--train_split='train'

Error message:
Gradient for xception_65/entry_flow/block3/unit_1/xception_module/separable_conv2_depthwise/BatchNorm/gamma:0 is NaN : Tensor had NaN values

Let use know if we are missing any parameter or any other issue.

Thanks,
chandra
